apps/common/rest_framework/mixins.py

```diff
@@ -35,6 +39,8 @@ def get_serializer_class_by_action(self, action: str):
             default_class = self.serializer_classes.get("write", default_class)
             if action == "partial_update":
                 default_class = self.serializer_classes.get("update", default_class)
+        if action == "destroy":
+            default_class = DefaultDestroySerializer
```

Comment:
What does this do, especially in relation to Swagger?

Reply:
What the automatic schema generation does is artificially calling the API endpoints to figure auto how they work. When they call the endpoints with the `destroy` action it tries to get the serializer for it.

The problem is that destroy action doesn't actually use a serializer. But for swagger the function for getting the serializer is used anyways, in case someone decides to modify their `destroy` action to use a serializer.

This usually works when a view has defined a single serializer, since it will just get that serializer. But when we define specific serializers for specific actions we usually don't define a serializer for the destroy action, since it is not used. This just sets a default for the schema view to have something to render. I'll add a comment :smile:

Reply:
I see! But won't this then create a lot of "rubbish" documentation for `delete`-methods which is not in use or which does not use the serializer? Or is this filtered away by Swagger?

Reply:
I actually don't know. Looking at the pages it seems to not be there, so I think its good?